tests/test_metric.py

```diff
@@ -478,6 +478,13 @@ def test_input_tf(self):
         self.assertDictEqual(expected_results, metric.compute())
         del metric
 
+    def test_string_casting(self):
+        metric = DummyMetric(experiment_id="test_string_casting")
+        metric.info.features = Features({"predictions": Value("string"), "references": Value("string")})
+        metric.compute(predictions=["a"], references=["a"])
+        with self.assertRaises(ValueError):
+            metric.compute(predictions=[1], references=[1])
+
```

Comment:
You can also add this one :)
```suggestion

        metric = DummyMetric(experiment_id="test_string_casting_2")
        metric.info.features = Features({"predictions": [Value("string")], "references": [Value("string")]})
        metric.compute(predictions=[["a"]], references=[["a"]])
        with self.assertRaises(ValueError):
            metric.compute(predictions=["a"], references=["a"])

```